src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
I don't get why we're coupling these components more if it doesn't produce some sort of concrete advantage. Risk of someone messing with this and breaking it seems super low regardless of the approach we use to code it, so why choose the design with more coupling, more pieces, more testing required, etc...